pkgs/shells/pdksh/default.nix

```diff
@@ -0,0 +1,38 @@
+{ stdenv, fetchurl }:
+
+with stdenv.lib;
+stdenv.mkDerivation rec {
+
+  name = "pdksh-${version}";
+  version = "5.2.14";
+
+  src = fetchurl {
+    url = "http://www.cs.mun.ca/~michael/pdksh/files/${name}.tar.gz";
+    sha256 = "19mzkfmj3ms8sp9m8f6fyba0am6fxnr538xw6b3fv4gja3fvq5db";
+  };
+
+  buildInputs = [ ];
+
+  # Traditionally, PDKSH is used as "pdksh" rather than "ksh"
+  configureFlags = [ "--program-prefix=pd" ];
+
+  patches = [
```

Comment:
These patches should really go into upstream, they look like behaviour-changing to me and I think we shouldn't patch software that much in a distribution repository!